contrib/syslog-debun

```diff
@@ -402,44 +449,87 @@ acquire_syslog_var () {
 	fi
 }
 
-acquire_syslog_ldinfo () {
-	ldd $syslogrealbin >${tmpdir}/syslog.ldd
+format_ldd_output () {
 	while read x ; do
 		x="${x#*=>}"
 		x="/${x#*/}"
 		x="${x%% (*}"
-		#[ "${x:0:2}" != "/ " ] && echo "$x" >>${tmpdir}/syslog.ldfiles
-		[ "${x}" != "/ " ] && echo "$x" >>${tmpdir}/syslog.ldfiles
-	done <${tmpdir}/syslog.ldd 
+		x="${x%%(*}"
+		[ "${x}" != "/ " ] && [ -f "${x}" ] && echo "$x"
+	done
+}
+
+
+acquire_syslog_ldinfo () {
+	$lddcmd $syslogrealbin |grep -v needs >${tmpdir}/syslog.ldd
+	format_ldd_output <${tmpdir}/syslog.ldd >${tmpdir}/syslog.ldfiles
 	for i in `cat ${tmpdir}/syslog.ldfiles ` ; do
 		distpkgoffile $i >>${tmpdir}/syslog.ldpkg
 	done
+	distpkgoffile_cleanup
 	sort <${tmpdir}/syslog.ldpkg | uniq >${tmpdir}/syslog.ldpkg.u
 	mv ${tmpdir}/syslog.ldpkg.u ${tmpdir}/syslog.ldpkg
 	for i in `cat ${tmpdir}/syslog.ldpkg ` ; do
 		distpkgstatus $i >>${tmpdir}/syslog.ldinfos
 	done
 }
 
+acquire_syslog_startup_method () {
+	echo "Detecting init system: "
+	if [ -d "/run/systemd/system" ]; then
+		acquire_syslog_startup_systemdunit
+	elif [ -d "/lib/svc/method" ]; then
+		acquire_syslog_startup_svc
+	else
+		acquire_syslog_startup_initscript
+	fi
+}
+
+acquire_syslog_startup_initscript () {
+	if [ -n "${initfile}" ]; then
+		echo "falling back to SystemV init style..."
+		cp "${initfile}" "${tmpdir}/sys.init.syslog-ng"
+		chmod 0660 "${tmpdir}/sys.init.syslog-ng"
+	else
+		echo "none."
+	fi
+}
+
+acquire_syslog_startup_systemdunit () {
+	echo "systemd detected..."
+	cp "/usr/lib/systemd/system/syslog-ng.service" "${tmpdir}/sys.service.syslog-ng"
+}
+
+acquire_syslog_startup_svc () {
+	echo "Solaris SMF detected..."
+	cp "/lib/svc/method/syslog-ng" "${tmpdir}/sys.svc-method.syslog-ng"
+	chmod 0660 "${tmpdir}/sys.svc-method.syslog-ng"
+	cp "/var/svc/manifest/system/syslog-ng.xml" "${tmpdir}/sys.svc-manifest.syslog-ng.xml"
+}
```

Comment:
Yes, your method file (the one that can be extended easily to handle multiple instances) is now being shipped with syslog-ng. This current version does in deed not check for multiple instances, but can't see this feature being implemented in this changeset.
I will put this on the wishlist as well for a future version.